src/uhlive/stream/conversation/client.py

```diff
@@ -163,6 +167,9 @@ def request_id(self) -> str:
     def receive(self, data: str) -> Event:
         """Decode received text frame"""
         event = Event.from_message(json.loads(data))
+        assert (
+            event.topic == self.topic
+        ), "Topic mismatch! Are you trying to mix several conversations on the same socket? You can't."
```

Comment:
This is not a `ProtocolError` proper, but rather an invariant violation, hence the `AssertionError` instead.